### PR TITLE
Merge ChanJianHao-Workspace-Delete

### LINKED ISSUE #145: Update SwitchWorkspace to switchws

### LINKED ISSUE #163: Merge and update Workspace command
Add delete workspace option for user. 
Combine existing workspace commands.

### PR DESCRIPTION
**Pull request by ChanJianHao**
Removed `addws `and `switchws`
Added `workspace `command with parameters -n, -s, -l, -d for create, switch, list, and delete respectively.

Closes #163

### CODE REVIEW

**Review by EyoWeiChin — APPROVED** [review]
LGTM! Thanks for refactoring into a single command.

File: src/main/java/anichan/Main.java [hunk 1]
```diff
@@ -107,14 +107,14 @@ public Main() {
...
             try {
+                Workspace newWorkspace = user.addWorkspace("Default");
+                ArrayList<Watchlist> watchlistList = new ArrayList<>();
+                watchlistList.add(new Watchlist("Default"));
+                newWorkspace.setWatchlistList(watchlistList);
+                user.setActiveWorkspace(newWorkspace);
+                LOGGER.log(Level.INFO, "Workspace created: " + newWorkspace);
+
                 storageManager.saveWorkspace(newWorkspace);
                 LOGGER.log(Level.INFO, "Workspace saved to storage: ");
```
Comment (OngDeZhi) [inline src/main/java/anichan/Main.java]: Is it intending to log some workspace information here after it is saved? There's a ":" at the end for line 119.

File: src/main/java/anichan/parser/WorkspaceParser.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                break;
+            case "d":
+                commandOption = "d";
+                break;
+            default:
+                LOGGER.log(Level.WARNING, "Invalid parameters provided");
+                throw new AniException(EXCEPTION_INVALID_PARAMETERS);
+            }
+
+            if (!commandOption.equals("l")) {
```
Comment (OngDeZhi) [inline src/main/java/anichan/parser/WorkspaceParser.java]: I think you could consider using a constant here.

File: src/main/java/anichan/storage/StorageManager.java [hunk 3]
```diff
@@ -58,6 +63,22 @@ public void saveWorkspace(Workspace workspace) throws AniException {
...
+
+    public void deleteWorkspace(String name) throws AniException {
+        String deletePathString = storageDirectory + name;
+        Path deletePath = Paths.get(deletePathString);
+
+        try {
+            Files.walk(deletePath)
+                    .sorted(Comparator.reverseOrder())
+                    .map(Path::toFile)
+                    .forEach(File::delete);
```
Comment (OngDeZhi) [inline src/main/java/anichan/storage/StorageManager.java:76]: Looks good!

**Review by OngDeZhi — CHANGES_REQUESTED** [review]
Just some minor amendments needed, otherwise, it looks good.

File: src/main/java/anichan/Main.java (hunk 1 above)
Comment (ChanJianHao) [reply src/main/java/anichan/Main.java]: Fixed that, good catch.

File: src/main/java/anichan/parser/WorkspaceParser.java (hunk 2 above)
Comment (ChanJianHao) [reply src/main/java/anichan/parser/WorkspaceParser.java]: Thanks for the reminder!

**Review by OngDeZhi — APPROVED** [review]
LGTM.

**Pull request merged**